### ISSUE TITLE
CI failure: lambdify: NameError: name 'polygamma' is not defined

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#23924]
This failure is seen in CI for the 3.11 optional dependency tests:
```
____________ sympy/utilities/tests/test_lambdify.py:test_scipy_fns _____________
Traceback (most recent call last):
  File "/home/runner/work/sympy/sympy/sympy/utilities/tests/test_lambdify.py", line 1103, in test_scipy_fns
    assert abs(f(tv) - sympy_result) < 1e-13*(1 + abs(sympy_result))
               ^^^^^
  File "<lambdifygenerated-604>", line 2, in _lambdifygenerated
    return polygamma(0, x)
           ^^^^^^^^^
NameError: name 'polygamma' is not defined
```
That CI job uses latest cython, numpy and scipy from git so this is likely due to a change in scipy.

**Comment by oscarbenjamin** [comment]
This job is not "required" so a PR can still be merged if it fails. Resolving the cause of this though is a release blocker for 1.11. If the next SciPy release is going to break `lambdify` then we might need a fix for that in SymPy 1.11.

I don't know `lambdify` very well though so I would appreciate it if someone who does takes a look at what happens with lambdifying polygamma using the latest scipy from git.

**Comment by oscargus** [comment]
Polygamma is still in the same location in SciPy if nothing else. https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.polygamma.html
